### DISCUSSION TITLE
Hide Table filters based on other filters value

### DISCUSSION THREAD

**Discussion by gsusanj** [filamentphp/filament#9193] [Help] [answered]
Is it possible to hide a filter, based on a value of another filter? For example I have polymorphic relationship filtering, and if filtering by a special type that other filter can not filter, I want to hide it.

I tried to use the Table $table->getFilters() or $table->getFilter('type') inside a ->visible() closure but that caused an 'ERR_CONNECTION_RESET' with no error in the log.

```
SelectFilter::make( 'my_filter )
    ->label('My Filter )
    ->options( [ ... ])
    ->visible(fn(Table $table) => dd($table->getFilters()))
```

**Comment by leandrocfe** [comment, marked as answer]
You could use [custom filter forms](https://filamentphp.com/docs/3.x/tables/filters#custom-filter-forms)
```php
->filters([
    Filter::make('filters')
        ->form([
            Select::make('filter_1')
                ->options([
                    'option_1' => 'Option 1',
                    'option_2' => 'Option 2',
                ]),
            Select::make('filter_2')
                ->options([
                    'option_3' => 'Option 3',
                    'option_4' => 'Option 4',
                ])
                ->hidden(fn (Get $get): bool => $get('filter_1') === 'option_1'),
        ])
])
```

**Reply by gsusanj** [reply]
Thanks, I have not considered using this.
